### PR TITLE
Add brave://crypto-wallets WebUI page

### PR DESCRIPTION
**Pull request by bbondy**
See the issue for a description of how it works

Resolves https://github.com/brave/brave-browser/issues/17996

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by mkarolin — APPROVED** [review]
`chromium_src` change ++

File: browser/ui/toolbar/brave_location_bar_model_delegate.cc [hunk 1]
```diff
@@ -38,15 +38,16 @@ void BraveLocationBarModelDelegate::FormattedStringFromURL(
...
         new_formatted_url, 0,
         base::UTF8ToUTF16(ethereum_remote_client_phishing_url),
-        u"brave://wallet");
+        u"brave://crypto-wallets");
     base::ReplaceFirstSubstringAfterOffset(
         new_formatted_url, 0,
         base::UTF8ToUTF16(ethereum_remote_client_ens_redirect_url),
-        u"brave://wallet");
+        u"brave://crypto-wallets");
   }
```
Comment (spylogsster) [inline browser/ui/toolbar/brave_location_bar_model_delegate.cc:51]: maybe something like this:
```

  bool replace_scheme_to_brave = url.SchemeIs("chrome");

#if BUILDFLAG(ETHEREUM_REMOTE_CLIENT_ENABLED)
  if (url.SchemeIs(kChromeExtensionScheme) &&
      url.host() == ethereum_remote_client_extension_id) {
    auto crypto_wallets = base::UTF8ToUTF16(kCryptoWalletsURL);
    base::ReplaceFirstSubstringAfterOffset(
        new_formatted_url, 0,
        base::UTF8ToUTF16(ethereum_remote_client_base_url),
        crypto_wallets);
    base::ReplaceFirstSubstringAfterOffset(
        new_formatted_url, 0,
        base::UTF8ToUTF16(ethereum_remote_client_phishing_url),
        crypto_wallets);
    base::ReplaceFirstSubstringAfterOffset(
        new_formatted_url, 0,
        base::UTF8ToUTF16(ethereum_remote_client_ens_redirect_url),
        crypto_wallets);
    replace_scheme_to_brave = true;
  }
#endif

  if (replace_scheme_to_brave) {
    base::ReplaceFirstSubstringAfterOffset(new_formatted_url, 0, u"chrome://",
                                           u"brave://");
  }
```

File: browser/ui/webui/brave_webui_source.cc [hunk 2]
```diff
@@ -408,7 +408,7 @@ void CustomizeWebUIHTMLSource(const std::string &name,
 #endif
       }
     }, {
-      std::string("wallet"), {
+      std::string("crypto-wallets"), {
```
Comment (spylogsster) [inline browser/ui/webui/brave_webui_source.cc:411]: kCryptoWalletsHost

File: chromium_src/chrome/browser/sessions/session_common_utils.cc [hunk 3]
```diff
@@ -21,7 +21,10 @@
 
 bool ShouldTrackURLForRestore(const GURL& url) {
 #if BUILDFLAG(ETHEREUM_REMOTE_CLIENT_ENABLED)
-  if (url.SchemeIs(content::kChromeUIScheme) && url.host() == "wallet") {
+  // TODO(bbondy): Once we have wallet live we should allow session restore here
+  // for brave://wallet
+  if (url.SchemeIs(content::kChromeUIScheme) &&
+      (url.host() == "crypto-wallets" || url.host() == "wallet")) {
```
Comment (spylogsster) [inline chromium_src/chrome/browser/sessions/session_common_utils.cc:27]: `kCryptoWalletsHost` and `kWalletPageHost`

**Comment by bbondy** [comment]
Sorry we decided to not go ahead with this plan and instead do this: https://github.com/brave/brave-browser/issues/17996
So closing this PR.

**Pull request closed without merging**